src/libstore/local-store.hh

```diff
@@ -46,6 +46,22 @@ struct LocalStoreConfig : virtual LocalFSStoreConfig
         "require-sigs",
         "Whether store paths copied into this store should have a trusted signature."};
 
+    Setting<bool> readOnly{(StoreConfig*) this,
+        false,
+        "read-only",
+        R"(
+          Allow this store to be opened when its [database](@docroot@/glossary.md#gloss-nix-database) is on a read-only filesystem.
+
+          Normally Nix will attempt to open the store database in read-write mode, even for querying (when write access is not needed).
+          This causes it to fail if the database is on a read-only filesystem.
```

Comment:
Minor confusion for me: "This causes it to fail" could as well mean "Enabling this option causes it to fail", which you probably don't mean.
```suggestion
          Normally Nix will attempt to open the store database in read-write mode, even for querying (when write access is not needed), causing it to fail if the database is on a read-only filesystem.
```